src/app/qgslayertreeviewprojecttimeindicator.h

```diff
@@ -0,0 +1,53 @@
+/***************************************************************************
+                         qgslayertreeviewprojecttimeindicator.h
+                         ---------------
+    begin                : February 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef QGSLAYERTREEVIEWPROJECTTIMEINDICATOR_H
+#define QGSLAYERTREEVIEWPROJECTTIMEINDICATOR_H
+
+
+#include "qgslayertreeviewindicatorprovider.h"
+
+/**
+ * Adds indicators for showing temporal layers that are using project time to set their
+ * current active temporal range.
+ */
+class QgsLayerTreeViewProjectTimeIndicatorProvider : public QgsLayerTreeViewIndicatorProvider
```

Comment:
@Samweli could you condense this logic into QgsLayerTreeViewTemporalIndicatorProvider ? It'll just be a matter of changing the return icon and tooltip text based on the layer's temporal source, so I don't think there's a strong need for two separate classes here.